solr/core/src/java/org/apache/solr/search/DocSetQuery.java

```diff
@@ -81,6 +81,9 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
         return new ConstantScoreWeight(this, boost) {
             @Override
             public Scorer scorer(LeafReaderContext context) {
+                if (docSet == null) {
```

Comment:
How could docSet be null?  I think it can't; and if I'm right we should maybe even use `Objects.requireNonNull ` in the constructor

Reply:
Thanks for suggesting this. It helped me learn more about `Objects.requireNonNull` and use cases.  🙂